service/db.go

```diff
@@ -14,52 +14,93 @@ import (
 	"github.com/spacemeshos/poet/shared"
 )
 
+type dataType byte
+
+const (
+	typeProof dataType = iota
+	typeRound
+)
+
 var ErrNotFound = leveldb.ErrNotFound
 
-type ProofsDatabase struct {
+type ServiceDatabase struct {
 	db     *leveldb.DB
 	proofs <-chan shared.ProofMessage
+	rounds <-chan RoundData
+}
+
+func makeKey(t dataType, k []byte) []byte {
+	key := make([]byte, len(k)+1)
```

Comment:
This should probably be

```suggestion
	key := make([]byte, 0, len(k)+1)
```

otherwise the returned `key` is `2*len(k) + 2` bytes long where the first `len(k)+1` bytes are all `0x00`

Reply:
Yes, thanks!